_source/_posts/2022-02-07-cookies-vs-tokens.adoc

```diff
@@ -0,0 +1,135 @@
+---
+layout: blog_post
+title: "A Comparison of Cookies and Tokens for Secure Authentication"
+author: teniola-fatunmbi
+by: contractor
+communities: [javascript]
+description: "This article examines the use of cookies vs. tokens for authentication, comparing the pros and cons of each method, so that you can determine which is best for your project."
+tags: [security, javascript, jwt, json-web-tokens, tokens, https, cookies]
+tweets:
+- "Trouble deciding between cookies or tokens? Here are some pros and cons of each method to capture your authentication data. "
+- "Cookies and tokens, apples and oranges. See which method fits your authentication use case the best?"
+image: blog/cookies-vs-tokens/cookies-vs-tokens-social.png
+type: awareness
+---
+:page-liquid:
+:toc: macro
+:experimental:
+
+Access control in websites and web applications is a top priority for security, but how you set it up depends on how you store the data to be authenticated.This, in turn, enables user authorization.Cookies and tokens are two common ways of setting up authentication..
+
+Cookies are chunks of data created by the server and sent to the client for communication purposes. Tokens, usually JSON Web Tokens (JWTs), are signed credentials encoded into a long string of characters created by the server.The main difference between cookies and tokens is their nature: tokens are stateless while cookies are stateful.
+
+With this in mind, why is there a need to store authentication on the browser? Because HTTP is stateless, even if you authenticate with one request, the server essentially "forgets" that authentication with subsequent requests. Therefore, you need to supply the token/cookie on every request for authentication by the server. The frontend stores the token or cookie and uses it to make subsequent requests to the server until the cookie or token is expired.
```

Comment:
```suggestion
With this in mind, why is there a need to store authentication on the browser? Because HTTP is stateless, even if you authenticate with one request, the server essentially "forgets" that authentication with subsequent requests. Therefore, you need to supply the token/cookie on every request for authentication by the server. The frontend stores the token or cookie and uses it to make subsequent requests to the server until the cookie or token expires.
```
active voice